src/sys/reboot.rs

```diff
@@ -0,0 +1,44 @@
+use {Errno, Error, Result};
+use libc;
+use void::Void;
+use std::mem::drop;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum RebootMode {
+    Halt,
+    kexec,
+    PowerOff,
+    Restart,
+    // we do not support Restart2,
+    Suspend,
+}
+
+pub fn reboot(how: RebootMode) -> Result<Void> {
+    let cmd = match how {
+        RebootMode::Halt => libc::RB_HALT_SYSTEM,
+        RebootMode::kexec => libc::RB_KEXEC,
+        RebootMode::PowerOff => libc::RB_POWER_OFF,
+        RebootMode::Restart => libc::RB_AUTOBOOT,
+        // we do not support Restart2,
+        RebootMode::Suspend => libc::RB_SW_SUSPEND,
+    };
+    unsafe {
+        libc::reboot(cmd)
+    };
+    Err(Error::Sys(Errno::last()))
```

Comment:
Errno::result(res).map(drop) should work here just like in `set_cad_enabled`, so that we consistently have `Result<()>` as the return value.